src/Native/Unix/System.Native/pal_networking.c

```diff
@@ -2220,6 +2221,19 @@ static int32_t TryChangeSocketEventRegistrationInner(
                0,
                0,
                GetKeventUdata(data));
+#if defined(__FreeBSD__)
```

Comment:
could it be feature detection in cmake instead of platform detection (which will unnecessarly require to change this line for NetBSD, DragonFlyBSD, Solaris etc. in future)? There already are `KEVENT` related `HAVE`s pragmas defined in corefx, e.g. https://github.com/dotnet/corefx/blob/b15f5097dfd600a6932e1b011970e4a09367af4a/src/Native/Unix/Common/pal_config.h.in#L29

https://github.com/dotnet/corefx/issues/2046 (mono code link at the bottom)

Reply:
@kasper3 I think that the reason for making this FreeBSD specific for now is that @wfurt found it as a problem on FreeBSD. We have no idea if it is FreeBSD specific or if it is a general problem of the kqueue with sockets. 
So we can either keep this change under the ifdef or make it unconditional. If we keep the ifdef, we would discover right after someone tries it on some other system with kqueue support whether it was FreeBSD specific or a general issue. If we make it unconditional, we pay the price of two syscalls instead of one even on systems where this is not needed (if there are any).